UE4SS/src/GUI/LiveView.cpp

```diff
@@ -279,6 +280,137 @@ namespace RC::GUI
+        auto& json_filters = json.new_array(STR("Filters"));
+        {
+            add_bool_filter_to_json(json_filters, STR("IncludeInheritance"), LiveView::s_include_inheritance);
+            add_bool_filter_to_json(json_filters, STR("UseRegexForSearch"), LiveView::s_use_regex_for_search);
+            add_bool_filter_to_json(json_filters, STR("ApplySearchFiltersWhenNotSearching"), LiveView::s_apply_search_filters_when_not_searching);
+            add_bool_filter_to_json(json_filters, Filter::DefaultObjectsOnly::s_debug_name, Filter::DefaultObjectsOnly::s_enabled);
+            add_bool_filter_to_json(json_filters, Filter::IncludeDefaultObjects::s_debug_name, Filter::IncludeDefaultObjects::s_enabled);
+            add_bool_filter_to_json(json_filters, Filter::InstancesOnly::s_debug_name, Filter::InstancesOnly::s_enabled);
+            add_bool_filter_to_json(json_filters, Filter::NonInstancesOnly::s_debug_name, Filter::NonInstancesOnly::s_enabled);
+        }
+        {
+            add_array_filter_to_json(json_filters, Filter::ClassNamesFilter::s_debug_name, Filter::ClassNamesFilter::list_class_names, STR("ClassNames"));
+            add_array_filter_to_json(json_filters, Filter::HasProperty::s_debug_name, Filter::HasProperty::list_properties, STR("Properties"));
+            add_array_filter_to_json(json_filters, Filter::HasPropertyType::s_debug_name, Filter::HasPropertyType::list_property_types, STR("PropertyTypes"));
+        }
+
+        auto json_file = File::open(
+                                    StringType{UE4SSProgram::get_program().get_working_directory()} + std::format(STR("\\liveview\\filters.meta.json")),
+                                    File::OpenFor::Writing,
+                                    File::OverwriteExistingFile::Yes,
+                                    File::CreateIfNonExistent::Yes);
+        int32_t json_indent_level{};
+        json_file.write_string_to_file(json.serialize(JSON::ShouldFormat::Yes, &json_indent_level));
+    }
+
+    static auto save_filters_to_disk() -> void
+    {
+        TRY([] {
+            internal_save_filters_to_disk();
+        });
+    }
+
+    static auto internal_load_filters_from_disk() -> void {
+        const auto json_file = File::open(
+            StringType{UE4SSProgram::get_program().get_working_directory()} + std::format(STR("\\liveview\\filters.meta.json")),
+            File::OpenFor::Reading,
+            File::OverwriteExistingFile::No,
+            File::CreateIfNonExistent::Yes
+        );
+        auto json_file_contents = json_file.read_all();
+        if (json_file_contents.empty()) {
+            return;
+        }
+
+        const auto json_global_object = JSON::Parser::parse(json_file_contents);
+        const auto& json_filters = json_global_object->get<JSON::Array>(STR("Filters"));
+        json_filters.for_each([&](const JSON::Value& filter) -> LoopAction {
+            auto& json_object = *filter.as<JSON::Object>();
```

Comment:
```suggestion
            if (!filter.is<JSON::Object>()) {
                throw std::runtime_error{"Invalid filter in 'filters.meta.json'"};
            }
            auto& json_object = *filter.as<JSON::Object>();
```
Again not asking you to make any changes since you've probably copied the code for watches, I'm just making you aware of how the API works.

The `as<T>` function makes an assumption that you've already confirmed that the type stored is actually `T`.
Your code doesn't confirm this.
It will still work but it will crash if this file is edited (externally from UE4SS) and the object is no longer an object.
Manual user edits or other external edits aside, what happens if we change the format of this file later ?
It will crash because we assumed T == Object but in the new format it might not be an object.

The `get<T>` function also works the same way.

My code suggestion shows how you would confirm the type.
I chose this particular bit of code for this example but there's plenty of places where this happens, just fyi.

Reply:
Thanks for explaining, I understand that now.